ndcube/visualization/mpl_plotter.py

```diff
@@ -145,7 +145,9 @@ def _plot_1D_cube(self, wcs, axes=None, axes_coordinates=None, axes_units=None,
             # We plot against pixel coordinates
             axes.errorbar(np.arange(len(ydata)), ydata, yerr=yerror, **kwargs)
         else:
-            axes.plot(ydata, **kwargs)
+            # compute() will break on numpy arrays to catch dask arrays by casting to bool instead
+            if not bool(np.isnan(ydata).all()):
```

Comment:
Could something like
```suggestion
            if np.isfinite(ydata).any() and unmasked:
```
be used instead without causing dask problems?  This way both NaN and Inf would be checked.  In addition, it would check whether there are valid unmasked data because  `unmasked` would be a `bool` that would indicate that at least one finite data value is unmasked and default to `True`.  It would be calculated around [here](https://github.com/sunpy/ndcube/pull/802/files#diff-87dd7a503ae11f67b56b44df5c6bbdb672ba9a78281c43669bc3763ece62200bR139).

Reply:
We will leave this up for the future